cloudwatch_logs_common/test/log_batch_test.cpp

```diff
@@ -0,0 +1,208 @@
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(1);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(2);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(ONE_DAY_IN_SEC+2);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(ONE_DAY_IN_SEC+1);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(ONE_DAY_IN_SEC);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  file_manager.write(log_data);
+
+  //read the batch
+  auto batch = file_manager.readBatch(test_strategy->logs.size());
+
+  //only the latest logs should be included in batch
+  ASSERT_EQ(3u, batch.batch_size);
+
+  //iterate through the logs in batch
+  auto it = batch.batch_data.begin();
+
+  //validate that they are the latest timestamps
+  ASSERT_EQ(ONE_DAY_IN_SEC, (*it).GetTimestamp());
+  it++;
+  ASSERT_EQ(ONE_DAY_IN_SEC+1, (*it).GetTimestamp());
+  it++;
+  ASSERT_EQ(ONE_DAY_IN_SEC+2, (*it).GetTimestamp());
+}
+TEST(log_batch_test, ALLPASS) {
+  //use test_strategy to mock read/write functions from data_manager_strategy
+  std::shared_ptr<TestStrategy> test_strategy = std::make_shared<TestStrategy>();
+  LogFileManager file_manager(test_strategy);
+  LogCollection log_data;
+  Aws::CloudWatchLogs::Model::InputLogEvent input_event;
+
+  //add test data to logs
+  input_event.SetTimestamp(0);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(1);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(2);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(3);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(4);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  input_event.SetTimestamp(ONE_DAY_IN_SEC-1);
+  input_event.SetMessage("Testing readBatch");
+  log_data.push_back(input_event);
+  file_manager.write(log_data);
+
+  //read the batch
+  auto batch = file_manager.readBatch(test_strategy->logs.size());
+
+  //only the latest logs should be included in batch
+  ASSERT_EQ(6u, batch.batch_size);
+
+  //iterate through the logs in batch
+  auto it = batch.batch_data.begin();
+
+  //validate that they are the latest timestamps
+  ASSERT_EQ(0, (*it).GetTimestamp());
+  it++;
+  ASSERT_EQ(1, (*it).GetTimestamp());
+  it++;
+  ASSERT_EQ(2, (*it).GetTimestamp());
+  it++;
+  ASSERT_EQ(3, (*it).GetTimestamp());
+  it++;
+  ASSERT_EQ(4, (*it).GetTimestamp());
+  it++;
+  ASSERT_EQ(ONE_DAY_IN_SEC-1, (*it).GetTimestamp());
+}
+TEST(log_batch_test, ONEPASS) {
```

Comment:
Can you follow the test naming convention used in the package? E.g. `this_is_my_test`